diff --git a/packages/flutter_app_packager/lib/src/makers/appimage/app_package_maker_appimage.dart b/packages/flutter_app_packager/lib/src/makers/appimage/app_package_maker_appimage.dart
index 4b4f7a7..a69a663 100644
--- a/packages/flutter_app_packager/lib/src/makers/appimage/app_package_maker_appimage.dart
+++ b/packages/flutter_app_packager/lib/src/makers/appimage/app_package_maker_appimage.dart
@@ -250,7 +250,7 @@ class AppPackageMakerAppImage extends AppPackageMaker {
           makeConfig.outputFile.path.replaceAll('.appimage', '.AppImage'),
         ],
         environment: {
-          'ARCH': 'x86_64',
+          'ARCH': _getArchitecture(),
         },
       ).then((value) {
         if (value.exitCode != 0) {
@@ -265,4 +265,13 @@ class AppPackageMakerAppImage extends AppPackageMaker {
       throw MakeError(e.toString());
     }
   }
+
+  String _getArchitecture() {
+    final result = Process.runSync('uname', ['-m']);
+    if ('${result.stdout}'.trim() == 'aarch64') {
+      return 'aarch64';
+    } else {
+      return 'x86_64';
+    }
+  }
 }
diff --git a/packages/flutter_app_packager/lib/src/makers/rpm/app_package_maker_rpm.dart b/packages/flutter_app_packager/lib/src/makers/rpm/app_package_maker_rpm.dart
index a25bb7e..bda362f 100644
--- a/packages/flutter_app_packager/lib/src/makers/rpm/app_package_maker_rpm.dart
+++ b/packages/flutter_app_packager/lib/src/makers/rpm/app_package_maker_rpm.dart
@@ -72,7 +72,7 @@ class AppPackageMakerRPM extends AppPackageMaker {
     final buildRoot = path.join(buildPath, makeConfig.appName);
     final specsPath = path.join(rpmbuildDirPath, 'SPECS');
     final rpmPath =
-        path.join(rpmbuildDirPath, 'RPMS', makeConfig.buildArch ?? 'x86_64');
+        path.join(rpmbuildDirPath, 'RPMS', makeConfig.buildArch ?? _getArchitecture());
     final buildWivesDirFile = Directory(buildRoot);
     if (!buildWivesDirFile.existsSync()) {
       buildWivesDirFile.createSync(recursive: true);
@@ -178,4 +178,13 @@ class AppPackageMakerRPM extends AppPackageMaker {
     packagingDirectory.deleteSync(recursive: true);
     return MakeResult(makeConfig);
   }
+
+  String _getArchitecture() {
+    final result = Process.runSync('uname', ['-m']);
+    if ('${result.stdout}'.trim() == 'aarch64') {
+      return 'aarch64';
+    } else {
+      return 'x86_64';
+    }
+  }
 }
diff --git a/packages/flutter_app_packager/lib/src/makers/rpm/make_rpm_config.dart b/packages/flutter_app_packager/lib/src/makers/rpm/make_rpm_config.dart
index 22d7215..0811a38 100644
--- a/packages/flutter_app_packager/lib/src/makers/rpm/make_rpm_config.dart
+++ b/packages/flutter_app_packager/lib/src/makers/rpm/make_rpm_config.dart
@@ -118,7 +118,7 @@ class MakeRPMConfig extends MakeConfig {
           'URL': url,
           'Requires': requires?.join(', '),
           'BuildRequires': buildRequires?.join(', '),
-          'BuildArch': buildArch ?? 'x86_64',
+          'BuildArch': buildArch ?? _getArchitecture(),
         }..removeWhere((key, value) => value == null),
         'body': {
           '%description': description ?? pubspec.description,
@@ -200,6 +200,15 @@ class MakeRPMConfig extends MakeConfig {
     };
     return Map.castFrom<String, String?, String, String>(map);
   }
+
+  String _getArchitecture() {
+    final result = Process.runSync('uname', ['-m']);
+    if ('${result.stdout}'.trim() == 'aarch64') {
+      return 'aarch64';
+    } else {
+      return 'x86_64';
+    }
+  }
 }
 
 class MakeRpmConfigLoader extends DefaultMakeConfigLoader {
